### PR TITLE
Improved branch coverage data package

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Improved branch coverage in data package to about 95%

### CODE REVIEW

File: src/main/java/data/Camera.java [hunk 1]
```diff
@@ -116,4 +110,35 @@ public void resetMovementMargin() {
     public static void incrementCounter() {
         instanceCounter++;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```
Comment (martijnjanssen) [inline src/main/java/data/Camera.java]: I think it would be better to use `! o instanceof Camera`.

File: src/main/java/data/Camera.java [hunk 2]
```diff
@@ -116,4 +110,35 @@ public void resetMovementMargin() {
...
+            return false;
+        }
+
+        Camera camera = (Camera) o;
+
+        if (Double.compare(camera.getMovementMargin(), getMovementMargin()) != 0) {
```
Comment (martijnjanssen) [inline src/main/java/data/Camera.java:125]: I am not a fan of if statements in an if statement. It's kinda weird.

File: src/main/java/data/Instrument.java [hunk 3]
```diff
@@ -39,4 +37,22 @@ public Instrument clone() {
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Instrument that = (Instrument) o;
+
+        if (getName() != null ? !getName().equals(that.getName()) : that.getName() != null) {
```
Comment (geenen124) [inline src/main/java/data/Instrument.java]: Please extract this conditional from the if

**Comment by martijnjanssen** [comment]
Good stuff, some small code style issues.

**Comment by Markvanderuit** [comment]
Looks very good. Glad to see branch coverage so high.

**Pull request merged**